File: dynamic_listings/C++/default/collisiondetection/circle_circle.cpp
#include <cmath>

struct Circle{
    // Let's define a circle class/structure
    Point center;
    int radius;
};

float distance(Point A, Point B){
    // Calculates the distance between two points
    return std::sqrt(std::pow((A.x - B.x),2) + std::pow((A.y - B.y),2));
}

bool circle_circle_collision(Circle A, Circle B){
    return distance(A.center, B.center) <= A.radius + B.radius;
}

File: dynamic_listings/rust/default/collisiondetection/circle_circle.rs
struct Circle {
    // Let's define a circle class/structure
    center: Point,
    radius: f32,
}

fn distance(a: &Point, b: &Point) -> f32 {
    // Calculates the distance between two points
    ((a.x - b.x).powi(2) + (a.y - b.y).powi(2)).sqrt()
}

fn circle_circle_collision(a: &Circle, b: &Circle) -> bool {
    distance(&a.center, &b.center) <= a.radius + b.radius
}
